### PR TITLE
mention routing from the database

### PR DESCRIPTION
**Pull request by dbu**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.3+ |
| Fixed tickets | #2186, relates to #5709 |

if we merge #6020, we might want to put this as subsection of the extending_routing section

### CODE REVIEW

File: cookbook/routing/routing_from_database.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+* Routes are stored in a database;
+* There is a database index on the path field, the lookup scales to huge
+  numbers of different routes;
+* Writes only affect the index of the database, which is very efficient;
+
+When all routes are known during deploy time and the number is not too
+high, using a :doc:`custom route loader <custom_route_loader>` is the
+preferred way to add more routes. When working with just one type of
+objects, a slug parameter on the object and the `@ParamConverter`
```
Comment (javiereguiluz) [inline cookbook/routing/routing_from_database.rst]: `@ParamConverter` should be enclosed with double backticks.

**Comment by dupuchba** [comment]
nice :-)

File: cookbook/routing/routing_from_database.rst (hunk 1 above)
Comment (dbu) [inline cookbook/routing/routing_from_database.rst]: ah darn. fixed.

File: cookbook/routing/index.rst [hunk 2]
```diff
@@ -12,3 +12,4 @@ Routing
     custom_route_loader
     redirect_trailing_slash
     extra_information
+    routing_from_database
```
Comment (xabbuh) [inline cookbook/routing/index.rst]: You also need to add a reference to `/cookbook/routing/map.rst.inc`.

File: cookbook/routing/routing_from_database.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+readable URL (e.g. for search engine optimization purposes), the routing
+can get slowed down. Additionally, if routes need to be edited by users,
+the route cache would need to be rebuilt frequently.
+
+For these cases, the DynamicRouter offers an alternative approach:
+
+* Routes are stored in a database;
+* There is a database index on the path field, the lookup scales to huge
+  numbers of different routes;
+* Writes only affect the index of the database, which is very efficient;
```
Comment (xabbuh) [inline cookbook/routing/routing_from_database.rst]: The last item should end with a dot.

File: cookbook/routing/routing_from_database.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+The core Symfony routing system is excellent at handling complex sets
+of routes. A highly optimized routing cache is dumped during
+deployments.
+
+However, when working with large amounts of data that each need a nice
+readable URL (e.g. for search engine optimization purposes), the routing
+can get slowed down. Additionally, if routes need to be edited by users,
+the route cache would need to be rebuilt frequently.
+
+For these cases, the DynamicRouter offers an alternative approach:
```
Comment (xabbuh) [inline cookbook/routing/routing_from_database.rst]: `DynamicRouter` should be enclosed with double backticks.

File: cookbook/routing/routing_from_database.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  numbers of different routes;
+* Writes only affect the index of the database, which is very efficient;
+
+When all routes are known during deploy time and the number is not too
+high, using a :doc:`custom route loader <custom_route_loader>` is the
+preferred way to add more routes. When working with just one type of
+objects, a slug parameter on the object and the ``@ParamConverter``
+annotation work fine (see FrameworkExtraBundle_) .
+
+What the DynamicRouter offers is routes with the full feature set of Symfony,
```
Comment (xabbuh) [inline cookbook/routing/routing_from_database.rst]: same here

File: cookbook/routing/routing_from_database.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
+.. index::
+   single: Routing; Extra Information
+
+Looking up Routes from a Database: Symfony CMF DynamicRouter
+============================================================
+
+The core Symfony routing system is excellent at handling complex sets
```
Comment (WouterJ) [inline cookbook/routing/routing_from_database.rst]: I think we're used to capitialize component names: "Symfony Routing system"

File: cookbook/routing/index.rst (hunk 2 above)
Comment (dbu) [inline cookbook/routing/index.rst]: done

**Comment by dbu** [comment]
i updated to all the feedback. anything left to do?

**Comment by dbu** [comment]
ping @weaverryan is there a reason to not merge this? happy to correct things if you can tell me what i should change.

**Comment by weaverryan** [comment]
@dbu no reason at all - sorry for the delay :)

**Pull request merged**